File: core/src/zyzxdev/swtch/util/ColorAccessor.java
package zyzxdev.swtch.util;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.math.Vector3;

import aurelienribon.tweenengine.TweenAccessor;

/**
 * Created by aaron on 5/30/17.
 */

public class ColorAccessor implements TweenAccessor<Color>{
	public static final int TYPE_RGB = 1;

	@Override
	public int getValues(Color target, int tweenType, float[] returnValues) {
		switch (tweenType) {
			case TYPE_RGB:
				returnValues[0] = target.r;
				returnValues[1] = target.g;
				returnValues[2] = target.b;
				return 3;
			default:
				assert false;
				return -1;
		}
	}

	@Override
	public void setValues(Color target, int tweenType, float[] newValues) {
		switch (tweenType) {
			case TYPE_RGB:
				target.r = newValues[0];
				target.g = newValues[1];
				target.b = newValues[2];
				break;
			default:
				assert false;
				break;
		}
	}
}

File: core/src/qixl/swtch/util/ColorAccessor.kt
package qixl.swtch.util

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.math.Vector3

import aurelienribon.tweenengine.TweenAccessor

/**
 * Created by aaron on 5/30/17.
 */

class ColorAccessor : TweenAccessor<Color> {

    override fun getValues(target: Color, tweenType: Int, returnValues: FloatArray): Int {
        when (tweenType) {
            TYPE_RGB -> {
                returnValues[0] = target.r
                returnValues[1] = target.g
                returnValues[2] = target.b
                return 3
            }
            else -> {
                assert(false)
                return -1
            }
        }
    }

    override fun setValues(target: Color, tweenType: Int, newValues: FloatArray) {
        when (tweenType) {
            TYPE_RGB -> {
                target.r = newValues[0]
                target.g = newValues[1]
                target.b = newValues[2]
            }
            else -> assert(false)
        }
    }

    companion object {
        val TYPE_RGB = 1
    }
}
